From d63e51354c2d5800043c906b34d007b8fa1229b0 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (Red Hat)" <rostedt@goodmis.org>
Date: Wed, 3 Jul 2013 13:32:37 -0400
Subject: [PATCH] trace-cmd: Make _GNU_SOURCE always defined

CFLAGS may be overridden by environment variables, which is fine as
not everyone wants -g -Wall. But the CFLAGS also incorporates the defining
of _GNU_SOURCE which is required to compile some parts of trace-cmd.
If CFLAGS is overridden but does not include _GNU_SOURCE, then the
compile will fail.

The solution is to always force _GNU_SOURCE to be defined by CFLAGS.

Reported-by: Florian Fainelli <f.fainelli@gmail.com>
Tested-by: Florian Fainelli <f.fainelli@gmail.com>
Signed-off-by: Steven Rostedt <rostedt@goodmis.org>
---
 Makefile | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6e6a183..51c6f39 100644
--- a/Makefile
+++ b/Makefile
@@ -218,9 +218,12 @@ INCLUDES = -I. $(CONFIG_INCLUDES)
 include features.mk
 
 # Set compile option CFLAGS if not set elsewhere
-CFLAGS ?= -g -Wall -D_GNU_SOURCE
+CFLAGS ?= -g -Wall
 LDFLAGS ?=
 
+# Required CFLAGS
+CFLAGS += -D_GNU_SOURCE
+
 ifndef NO_PTRACE
 ifneq ($(call try-cc,$(SOURCE_PTRACE),),y)
 	NO_PTRACE = 1
-- 
1.8.3.2

